lib/timeline/fps.dart

```diff
@@ -89,31 +89,45 @@ class FramesChart extends LineChart<FramesTracker> {
 }
 
 class FramesTracker {
-  FramesTracker(this.service) {
-    service.onExtensionEvent.listen((Event e) {
-      if (e.extensionKind == 'Flutter.Frame') {
-        final ExtensionData data = e.extensionData;
-        addSample(FrameInfo.from(data.data));
-      }
-    });
-  }
+  FramesTracker(this.service);
 
   static const int kMaxFrames = 60;
-
-  VmServiceWrapper service;
   final StreamController<Null> _changeController =
       StreamController<Null>.broadcast();
+
+  VmServiceWrapper service;
+  StreamSubscription<Event> eventStreamSubscription;
   List<FrameInfo> samples = <FrameInfo>[];
 
   bool get hasConnection => service != null;
 
   Stream<Null> get onChange => _changeController.stream;
 
-  void start() {}
+  void start() {
+    eventStreamSubscription = service.onExtensionEvent.listen((Event e) {
```

Comment:
make sure the subscription doesn't already exist.

Reply:
Added an assert.

Reply:
You might want to be defensive and even cancel the existing subscription if there was already one.

Reply:
done.